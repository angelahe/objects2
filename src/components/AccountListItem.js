import React from 'react';
import '../styles/Accounts.css';
import editbtn from '../images/edit_FFFFFF.png';
import deletebtn from '../images/delete_FFFFFF.png';

const AccountListItem = (props) => {

    function handleAccountClick(e) {
        console.log("in handleclick function generic");
        console.log("account index is ", e.target.parentNode.getAttribute("idindex"));
        console.log("account index 2 is", e.target.parentNode.getAttribute("idindex"));
        console.log("classname is ", e.target.parentNode.getAttribute("className"));
        const acctIndex = e.target.getAttribute("idindex");
        if(e.target.parentNode.getAttribute("className") === "AccountBtn Add")
            console.log("add was clicked");
        else if (e.target.parentNode.getAttribute("className") === "AccountBtn Delete")
            console.log("delete was clicked");
        else {
            //props.accountClicked(acctIndex);
            console.log("account item was clicked");
        }
    }
    function handleEditClick(e) {
        console.log("in handleEditClick of AccountListItem");
        console.log("event is", e);
        console.log("event target is", e.target);
        console.log("event target parent is ", e.target.parentNode)
        console.log("event target parent parent is ", e.target.parentNode.parentNode);
        console.log("index is ", e.target.parentNode.parentNode.getAttribute("idindex"));
        console.log("props is ", props);
        const acctIndex = e.target.parentNode.parentNode.getAttribute("idindex");
        props.editClicked(acctIndex);
        //props.onEditClick(acctIndex);
        //editClicked(acctIndex);
    }

    function handleDeleteClick(e) {
        console.log("in deleteclick");
        const acctIndex = e.target.parentNode.getAttribute("idindex");
        //props.deleteClicked(acctIndex);
        //props.onDeleteClick(acctIndex)
    }



    return (
        <div className="ItemBox AccountListItem" idindex={props.item.acctId}
             onClick={handleAccountClick}>
            <span className="DetailText">{props.item.acctName + " $" + props.item.balance}</span>
            <button className="AccountBtn Add" onClick={handleEditClick}>
                <img className="btnImg" src={editbtn} alt="Add"/>
            </button>
            <button className="AccountBtn Delete" onClick={handleDeleteClick}>
                <img className="btnImg" src={deletebtn} alt="Add"/>
            </button>
            <hr/>
        </div>

    )
}

export default AccountListItem;