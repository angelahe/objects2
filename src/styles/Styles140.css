.AppContainer {
    background-color: #e6f5ff;
    display: flex;
    flex-direction: row;
    justify-content: flex-start;

}

.AppPanel {
    background-color: whitesmoke;
    border-radius: 10px;
    border-width: 2px;
    border-color: #1aa3ff;
    margin: 10px;
    height: auto;
    width: 40%;
    justify-content: flex-start;

}
.Header {
    background-color: #1aa3ff;
    border-radius: 10px;
}

.AppBtn {
    color: white;
    height: 35px;
    width:35px;
    padding:0px;
    align-content: flex-start;
    border-radius: 5px;
    margin-left: 10px;

}
.ActionBox {
    height: auto;
    width: 100%;
    border-radius: 10px;

}
.ItemBox {
    border-radius: 10px;
    padding: 10px;
}
.AppHeader {
    background-color: #1aa3ff;
    display:flex;
    justify-content:space-between;
}

.AppList {
    width: 96%;
    padding: 5px;
}

.AppListItem {
    background-color: #e6f5ff;
}

.AppListItem:active {
    background-color: #ccebff;
    justify-content: space-around;
}

.AddItem {
    /*width: 50%;*/
    font-weight: bold;
    font-size: 1.5em;
    justify-content: flex-start;
    color: white;
    background-color: #1aa3ff;
    text-align: left;

}
.btnImg {
    height: 35px;
    width: 35px;
    background-color: #1aa3ff;
    margin:0;
}

.btnText {
    height: 35px;
    width: 200px;
    background-color: #1aa3ff;
    color: white;
    font-size: 1.5em;
    border-radius: 10px;

}

.DetailText {
    color: #1aa3ff;
    font-size: 1.5em;
    align-items: flex-start;
    justify-content: left;
    text-align: left;
    line-height: 1.5em;
}

.InputText {
    color: #1aa3ff;
    font-size: 1.5em;
}

.SummaryText {
    color: #1aa3ff;
    font-size: 1em;
    align-items: flex-start;
    text-align: left;
    margin-left: 10px;
    line-height: 1.5em;
}

hr {
    color: darkblue;
    height: 2px;
}

h2 {
    padding: 1px;
    color: #1aa3ff;
}

