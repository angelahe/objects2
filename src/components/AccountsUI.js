import React from 'react';
import Accounts from './accounts';
import Account from './account';
import AccountSummary from './AccountSummary';
import AccountCreate from './AccountCreate';
import AccountEdit from './AccountEdit';
import AccountDelete from './AccountDelete';
import AccountListItem from './AccountListItem';
import AccountActions from './AccountActions';
import addbtn from '../images/add_FFFFFF.png';

import '../styles/Accounts.css';


class AccountsUI extends React.Component {

    constructor(props) {
        super(props)
        this.state = {
            AccountList: new Accounts(),
            accountNumber: 0,
            nextId: 0,
            currentSelected: 0,
            currentAccount: "Savings",
            currentBalance: 0,
            lowestAccount: 0,
            highestAccount: 0,
            totalAccount: 0,
            addAccount: false,
            editShow: false,
            deleteShow: false,
            accountActionsShow: false
        }
    }

    onBtnAddClick = () => {
        const changeState = !this.state.addAccount;
        this.setState ({addAccount: changeState});
    }

    //make a copy of the current AccountList in state
    makeAccountCopy = () => {
        const AccountListCopy = new Accounts();
        for(let i=0; i<this.state.AccountList.Accounts.length; i++) {
            AccountListCopy.addAccount(this.state.AccountList.Accounts[i]);
        }
        return AccountListCopy;

    }

    onAccountCreate = (name, balance) => {
        const AccountListCopy = this.makeAccountCopy();
        AccountListCopy.addAccount(new Account(balance, name, this.state.nextId));

        this.setState({
            AccountList: AccountListCopy,
            accountNumber: AccountListCopy.Accounts.length,
            addAccount: !this.state.addAccount,
            nextId: this.state.nextId + 1
        });
    }

    onEditClick = (acctIndex) => {
        this.setState({
            editShow: !this.state.editShow,
            currentSelected: acctIndex
        });

    }

    onAccountUpdate = (name, balance, index) => {
        console.log("in account update with: ", name, balance, index);
        const updateWithThis = new Account(balance, name, index);
        const AccountListCopy = new Accounts();
        const changeState = !this.state.editShow;

        //make a copy of Accountlist, replacing the edited account
        for(let i=0; i<this.state.AccountList.Accounts.length; i++) {
            if(i === index)
                AccountListCopy.addAccount(updateWithThis);
            else
                AccountListCopy.addAccount(this.state.AccountList.Accounts[i]);
        }

        this.setState({
            AccountList: AccountListCopy,
            editShow: changeState
        });
    }

    onDeleteClick = (acctIndex) => {
        console.log("in delete click", acctIndex);
        this.setState({
            deleteShow: true,
            currentSelected: acctIndex
        });
    }

    onDeleteAccount = (acctIndex) => {
        console.log("in delete account", acctIndex);
        const AccountListCopy = new Accounts();

        for(let i=0; i<this.state.AccountList.Accounts.length; i++) {
            if (i!== acctIndex)
                AccountListCopy.addAccount(this.state.AccountList.Accounts[i]);
        }
        this.setState({
            AccountList: AccountListCopy,
            deleteShow: false,
            accountNumber: AccountListCopy.Accounts.length,
            currentSelected: 0
        });
    }

    onAccountClick = (acctIndex) => {
        console.log("in account click", acctIndex);
        this.setState({
            accountActionsShow: true,
            currentSelected: acctIndex
        })
    }

    onDepositToAccount = (amount, acctIndex) => {
        console.log("in onDepositToAccount");
        const AccountListCopy = new Accounts();
        for(let i=0; i<this.state.AccountList.Accounts.length; i++) {
            AccountListCopy.addAccount(this.state.AccountList.Accounts[i]);
        }
        AccountListCopy.depositToAccount(acctIndex, amount);
        this.setState({
            AccountList: AccountListCopy,
            accountActionsShow: false,
        });
    }

    onWithdrawFromAccount = (amount, acctIndex) => {
        console.log("in onWithdrawFromAccount for of ", amount, acctIndex);
        const AccountListCopy = new Accounts();
        for(let i=0; i<this.state.AccountList.Accounts.length; i++) {
            if (i!== acctIndex)
                AccountListCopy.addAccount(this.state.AccountList.Accounts[i]);
        }
        AccountListCopy.withdrawFromAccount(acctIndex, amount);
        this.setState({
            AccountList: AccountListCopy,
            accountActionsShow: false
        });
    }

    render() {

//        console.log("Account list items are ", this.state.AccountList);
        console.log("addAccount is ", this.state.addAccount);
        const accountListItems = this.state.AccountList.Accounts.map((item, index) =>
           <AccountListItem key={item.acctId}
                            index = {index}
                            item={item}
                            accountClicked={this.onAccountClick}
                            editClicked={this.onEditClick}
                            deleteClicked={this.onDeleteClick}
           />
        );
//<AccountSummary accounts={this.state.AccountList.Accounts} />

        return(
            <div>
                <h1>Hello from Accounts</h1>
                <div className = "AccountContainer">
                    <AccountSummary accounts={this.state.AccountList} />
                    <div className = "AccountPanel">
                          <div className = "ItemBox AccountHeader">
                            <span className = "AddAccount">Add Account</span>
                            <button className="AccountBtn" onClick={this.onBtnAddClick}>
                                <img className="btnImg" src={addbtn} alt="Add"/>
            `                    </button>
                          </div>
                        <div className = "AccountList">
                            {accountListItems}
                        </div>
                    </div>
                    <div className = "AccountPanel">
                        {(this.state.addAccount)
                            ? <AccountCreate {...this.state}
                                             createClicked = {this.onAccountCreate}
                              />
                            : null }
                        {(this.state.editShow)
                            ? <AccountEdit {...this.state}
                                         index= {this.state.currentSelected}
                                         updateAccount = {this.onAccountUpdate}
                              />
                            :null }
                        {(this.state.deleteShow)
                            ? <AccountDelete {...this.state}
                                           index = {this.state.currentSelected}
                                           deleteAccount = {this.onDeleteAccount}
                              />
                            :null }
                        {(this.state.accountActionsShow)
                            ? <AccountActions {...this.state}
                                              index={this.state.currentSelected}
                                              deposit={this.onDepositToAccount}
                                              withdraw={this.onWithdrawFromAccount}
                            />
                            : null
                        }
                    </div>

                </div>

            </div>
        )

    }
}
export default AccountsUI