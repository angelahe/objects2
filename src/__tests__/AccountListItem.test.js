import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import AccountsUI from "../components/AccountsUI";
import AccountListItem from "../components/AccountListItem";
import '../styles/Styles140.css'

import renderer from "react-test-renderer";

const props = new AccountsUI();
const item = {acctId : 1,
              acctName: "Savings",
              balance: 1000};

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<AccountListItem key={item.acctId}
                                   account={item}
                                   accountClicked={props.onAccountClick}
                                   editClicked={props.onEditClick}
                                   deleteClicked={props.onDeleteClick}
  />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders without crashing', () => {
  shallow(<AccountListItem key={item.acctId}
                           account={item}
                           accountClicked={props.onAccountClick}
                           editClicked={props.onEditClick}
                           deleteClicked={props.onDeleteClick}
  />);
});

test('AccountUI renders correctly', () => {
  const tree = renderer.create(<AccountListItem key={item.acctId}
                                                account={item}
                                                accountClicked={props.onAccountClick}
                                                editClicked={props.onEditClick}
                                                deleteClicked={props.onDeleteClick}

  />).toJSON();
  expect(tree).toMatchSnapshot();
});