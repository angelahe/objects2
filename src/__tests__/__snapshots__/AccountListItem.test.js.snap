// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountUI renders correctly 1`] = `
<div
  className="ItemBox AppListItem"
  elemtype="Container"
  onClick={[Function]}
>
  <span
    className="DetailText"
    elemtype="Text"
  >
    Savings $1000
  </span>
  <button
    className="AppBtn"
    elemtype="Edit"
  >
    <img
      alt="Add"
      className="btnImg"
      elemtype="Edit Image"
      src="edit_FFFFFF.png"
    />
  </button>
  <button
    className="AppBtn Delete"
    elemtype="Delete"
  >
    <img
      alt="Add"
      className="btnImg"
      elemtype="Delete Image"
      src="delete_FFFFFF.png"
    />
  </button>
  <hr />
</div>
`;
